frontend/src/network/create/create.html

```diff
@@ -4,7 +4,7 @@
         <div class="form-control">
             <input id="net-create-name" type="text" data-ng-model="networkCreateCtrl.form.Name" focus-me
                    data-required="required" name="name"
-                   data-ng-pattern="/^[A-Za-z0-9]+$/">
+                   data-ng-pattern="/^\w[\w_.-]+\w$/">
```

Comment:
Base on my test, the character ? do not match our purpose, but character * which means zero or more is much better matching the name format.